wgpu/src/util/mod.rs

```diff
@@ -191,3 +191,46 @@ pub fn pipeline_cache_key(adapter_info: &wgt::AdapterInfo) -> Option<String> {
         _ => None,
     }
 }
+
+/// Adds extra conversion functions to `TextureFormat`.
+pub trait TextureFormatExt {
+    /// Finds the [`TextureFormat`](wgt::TextureFormat) corresponding to the given
+    /// [`StorageFormat`](wgc::naga::StorageFormat).
+    ///
+    /// # Examples
+    /// ```
+    /// use wgpu::util::TextureFormatExt;
+    /// assert_eq!(wgpu::TextureFormat::from_storage_format(wgpu::naga::StorageFormat::Bgra8Unorm), wgpu::TextureFormat::Bgra8Unorm);
+    /// ```
+    #[cfg_attr(docsrs, doc(cfg(any(wgpu_core, naga))))]
+    #[cfg(any(wgpu_core, naga))]
+    fn from_storage_format(storage_format: crate::naga::StorageFormat) -> Self;
+
+    /// Finds the [`StorageFormat`](wgc::naga::StorageFormat) corresponding to the given [`TextureFormat`](wgt::TextureFormat).
+    /// Returns `None` if there is no matching storage format,
+    /// which typically indicates this format is not supported
+    /// for storage textures.
+    ///
+    /// # Examplees
+    /// ```
+    /// use wgpu::util::TextureFormatExt;
+    /// assert_eq!(wgpu::TextureFormat::Bgra8Unorm.to_storage_format(), Some(wgpu::naga::StorageFormat::Bgra8Unorm));
+    /// ```
+    #[cfg_attr(docsrs, doc(cfg(any(wgpu_core, naga))))]
```

Comment:
We shouldn't need this, we have https://github.com/gfx-rs/wgpu/blob/trunk/wgpu/src/lib.rs#L17 which should take care of it.